protos/block_and_transaction.proto

```diff
@@ -227,3 +227,12 @@ message ConsensusMetadata {
   // The Raft index associated with the block
   uint64 raft_index = 2;
 }
+
+message BlockTxIds {
+  repeated string tx_id = 1;
+}
+
+message AugmentedBlockHeader {
+  BlockHeader header = 1;
+  BlockTxIds tx_ids = 2;
```

Comment:
I would prefer to use the existing `BlockHeader` message type itself with the additional `repeated string tx_ids = 1`. The additional field would be empty when augmented is set to false. For marshalling, we can use some internal `BlockTxIds` proto at the `blockstore` package rather than using two `tx_ids`. In fact the `repeated string tx_id` should be `tx_ids`. I see it as a hack to avoid `GetTxIDs().GetTxIDs()`. Even the current usage of `GetTxIDs().GetTxID()[0]` is not good